settings/settings.go

```diff
@@ -80,6 +101,64 @@ func (upd *UpdateCheck) WriteToDisk() error {
 	return err
 }
 
+// Load will read the telemetry settings from the user's disk and then deserialize it into the current instance.
+func (tel *TelemetrySettings) Load() error {
+	path := filepath.Join(SettingsPath(), telemetryFilename())
+
+	if err := ensureSettingsFileExists(path); err != nil {
+		return err
+	}
+
+	content, err := FS.ReadFile(path) // #nosec
+	if err != nil {
+		return err
+	}
+
+	err = yaml.Unmarshal(content, &tel)
+	return err
+}
+
+// WriteToDisk will write the telemetry settings to disk by serializing the YAML
+func (tel *TelemetrySettings) Write() error {
+	enc, err := yaml.Marshal(&tel)
+	if err != nil {
+		return err
+	}
+
+	path := filepath.Join(SettingsPath(), telemetryFilename())
+	err = FS.WriteFile(path, enc, 0600)
+	return err
+}
+
+// Track takes a telemetry event, enrich with various data and sends it
+// This is the method you must use to send telemetry events
+// This will fail if 'checkTelemetry' has not called before
+func (cfg *Config) Track(event telemetry.Event) error {
```

Comment:
It feels a little strange the the `Track` method is part of the config, as it's not an action of the "configuration", but rather of the client. Should this go on the `Client`? Although I understand why it's easier to pluck the properties from the Config.

Now that I think of it, the weird part is having the `Client` on the `Config`. 🤔
Not sure you should change everything for this, but there's a possible bit of confusion here.